src/cargo/core/compiler/timings.rs

```diff
@@ -562,20 +709,53 @@ impl<'gctx> Timings<'gctx> {
       <th></th>
       <th>Unit</th>
       <th>Total</th>
-      <th>Codegen</th>
+      {headers}
```

Comment:
Would it work to split out a refactor commit that adjusts the existing Frontend/Backend logic to this and then add section timings on top?

Reply:
Hmm, it's all very dependent on one another, but I tried to split it.